Make Result.Interface() return an empty string, as is.

ozevren opened the pull request:
+ Removing the old handling code that converts an empty string to
nil.

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
NONE
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/istio/mixer/1523)
<!-- Reviewable:end -->

Commit:
Make Result.Interface() return an empty string, as is.

+ Removing the old handling code that converts an empty string to
nil.

ozevren commented:
Fixes https://github.com/istio/mixer/issues/934

istio-testing commented:
@ozevren: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/mixer-presubmit.sh | aeaba52d6ee806193ad3ac518d81273821638d06 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_mixer/1523/mixer-presubmit/1316/) | `/test mixer-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

ozevren commented:
/test mixer-presubmit

Test passes when run locally.

ozevren commented:
friendly ping

douglas-reid commented on pkg/il/compiler/compiler_test.go:
```diff
@@ -388,6 +388,20 @@ end
 `,
 	},
 	{
+		expr: `ar["b"]`,
```
won't this lead to the issue that originally caused us to make `""` => `nil` ?  what happens for: `ar["b"] | "other"`?

ozevren replied:
Depending on the bag contents, ar["b"] | "other" will evaluate to:
bag {} => "other"
bag { ar:{} } => "other"
bag { ar:{ "b": ""} } => ""

douglas-reid replied:
Do we have tests for those cases? If not, can we add them?

ozevren replied:
Yes we do. I've also added one more, just in case, in an additional changelist within the same PR.

Commit:
Bolster stringmap lookup & or tests (i.e. ar["c"] | "foo").

For expression: ar["c"] | "foo":

bag {} => "foo"
bag {"ar": {} } => "foo"
bag {"ar": { "c":"b" } => "b"

guptasu reviewed (approved):
LGTM

douglas-reid commented:
/hold

douglas-reid commented:
with the repo migration in-flight, we shouldn't merge any PRs at the moment.

ozevren commented:
What is the ETA for repo migration completion?

douglas-reid commented:
@ozevren i've been led to believe that it will be Wed. Nov. 1.  Any PRs submitted now would be lost in the shuffle. Unfortunately, this means you will have to move these changes over and resubmit (with hopefully a successful transfer of test infra, etc., already done and working). :(.

geeknoid commented:
This repo is no longer accepting PRs. Please resubmit this change to the istio/istio repo.

Thanks.
